elytron/src/test/java/org/wildfly/extension/elytron/RealmsTestCase.java

```diff
@@ -727,6 +728,65 @@ public void testPathToJAASConfigFileDoesNotExist() throws Exception {
         Assert.assertTrue(response.asString().contains("JAAS configuration file does not exist."));
     }
 
+    @Test
+    public void testDistributedRealm() throws Exception {
+        KernelServices services = super.createKernelServicesBuilder(new TestEnvironment()).setSubsystemXmlResource("realms-test.xml").build();
+        if (!services.isSuccessfulBoot()) {
+            Assert.fail(services.getBootError().toString());
+        }
+
+        ServiceName serviceName = Capabilities.SECURITY_REALM_RUNTIME_CAPABILITY.getCapabilityServiceName("DistributedRealmNoUnavailable");
+        SecurityRealm distributedRealm = (SecurityRealm) services.getContainer().getService(serviceName).getValue();
+        testDistributedRealmSuccessful(distributedRealm);
+
+        ServiceName serviceName2 = Capabilities.SECURITY_REALM_RUNTIME_CAPABILITY.getCapabilityServiceName("DistributedRealmFirstUnavailable");
+        SecurityRealm distributedRealm2 = (SecurityRealm) services.getContainer().getService(serviceName2).getValue();
+        testDistributedRealmFailure(distributedRealm2);
+
+        ServiceName serviceName3 = Capabilities.SECURITY_REALM_RUNTIME_CAPABILITY.getCapabilityServiceName("DistributedRealmFirstUnavailableIgnored");
+        SecurityRealm distributedRealm3 = (SecurityRealm) services.getContainer().getService(serviceName3).getValue();
+        testDistributedRealmSuccessful(distributedRealm3);
+
+        TestEnvironment.activateService(services, Capabilities.SECURITY_DOMAIN_RUNTIME_CAPABILITY, "DistributedRealmDomain");
+        ServiceName serviceName4 = Capabilities.SECURITY_DOMAIN_RUNTIME_CAPABILITY.getCapabilityServiceName("DistributedRealmDomain");
+        SecurityDomain domain = (SecurityDomain) services.getContainer().getService(serviceName4).getValue();
+        Assert.assertNotNull(domain);
+        if (SecurityDomain.getCurrent() == null) {
+            domain.registerWithClassLoader(Thread.currentThread().getContextClassLoader());
+        }
+        ServerAuthenticationContext context = domain.createNewAuthenticationContext();
+        context.setAuthenticationName("firstUser");
+        Assert.assertTrue(context.exists());
+
+        Assert.assertTrue(isSecurityRealmUnavailableEventLogged());
+    }
+
+    private boolean isSecurityRealmUnavailableEventLogged() throws Exception {
+        List<String> lines = Files.readAllLines(Paths.get("target/audit.log"), StandardCharsets.UTF_8);
+        for (String line : lines) {
+            if (line.contains("SecurityRealmUnavailableEvent")) {
+                return true;
```

Comment:
@lvydra Just a minor but can we check the name of the realm that threw the SecurityRealmUnavailableEvent as well just to know that the message is related to the event we want to test?